core/src/com/unciv/ui/worldscreen/bottombar/BattleTable.kt

```diff
@@ -236,11 +236,20 @@ class BattleTable(val worldScreen: WorldScreen): Table() {
         attackerNameWrapper.add(UnitGroup(attacker.unit,25f)).padRight(5f)
         attackerNameWrapper.add(attackerLabel)
         add(attackerNameWrapper)
-
+        var canNuke = true
         val defenderNameWrapper = Table()
         for (tile in targetTile.getTilesInDistance(Battle.NUKE_RADIUS)) {
             val defender = tryGetDefenderAtTile(tile, true)
             if (defender == null) continue
```

Comment:
Wow! That's much better!

However, it *will not work* in case, when the player is nuking an enemy territory without units.
As you see this `continue` will skip your check.